### DISCUSSION TITLE
PM2 Cron

### DISCUSSION THREAD

**Discussion by zernonia** [strapi/strapi#7513] [Legacy Topics]
Hi, I saw that `https://strapi.io/documentation/v3.x/guides/scheduled-publication.html#create-a-cron-task` mentioned that 

```
TIP

Please note that Strapi's built in CRON feature will not work if you plan to use pm2 or node based clustering. You will need to execute these CRON tasks outside of Strapi.
```

Since we cant use pm2 with Strapi-in-built cron task, any idea how can I trigger a task with Strapi.services using pm2? 
Thanks in advance!

**Comment by derrickmehaffy** [comment]
Hi @zernonia !

You are correct that using pm2 with Strapi's built in cron is not recommended (if you plan to use the pm2 clusters), a single instance without clusters will have no issues using the cron.

However my recommendation to you if you absolutely need to use a `strapi.services.*` would be to build a custom route (route/controller/service) that is protected and then simply use some other simple cron method to call this route when needed. This way you can properly scale out Strapi using pm2 clusters, or even on multiple servers and the cron task will only be executed by one node in the cluster, thus preventing any kind of race condition :smile: 

Where you execute the cron is entirely up to you, all linux operating systems offer a very simple crontasks option that can execute any commnd, bash script, or even python scripts. In this case you can simply setup a `curl` GET request with an authorization header (or just some custom token) on the Strapi route to ensure it's protected.

Alternatively you can create your own custom node script using something like [node-cron](https://www.npmjs.com/package/node-cron) that can also be ran in pm2 itself. (I do this on my own personal projects)

**Comment by zernonia** [comment]
@derrickmehaffy Thank you so much for your reply. I had the same idea about creating custom route, but I was hoping there's an easier way to link strapi cron to PM2. 

Or.. is there any alternative method rather than using PM2 to rebuild strapi when the server restarted? So I can utilize strapi-built-in cron.

**Reply by derrickmehaffy** [reply]
The issue itself isn't with PM2 but when you are running Strapi with PM2 clusters, the issue is mainly due to the fact that each instance isn't aware of the others so during a built-in cron all of the instances will trigger the cron and you hit a race condition.

I'm not sure what you mean with your 2nd question?
